add pull request template (#4315)
VictoriaMetrics/VictoriaMetrics: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md

### Describe Your Changes

Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.

### Checklist (Optional for External Contributions)

> - [ ] Include a link to the GitHub issue in the PR description, if possible.

Reviewer: PR description or commit message? I think it should be the latter.

> - [ ] Add an item to the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md) that describes the problem you're addressing and how you're solving it.

Reviewer: ```suggestion
- [ ] Mention the change in the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md). Explain what has changed and why. If there is a related issue or documentation change - link them as well.
```

> - [ ] Add an item to the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md) that describes the problem you're addressing and how you're solving it.

Reviewer: Please add the following:

```
- [ ] If the change somehow affects public interfaces (a new flag was added or updated, or some behavior has changed) - add the corresponding change to documentation.
```

>   Tips for writing a good changelog message::
> 
>     * Write a human-readable changelog message that describes the problem and solution.
>     * Include a link to the issue or pull request in your changelog message.
>     * Use specific language identifying the fix, such as an error message, metric name, or flag name.
>     * Provide a link to the relevant documentation for any new features you add or modify.

Reviewer: ```suggestion
```

I don't think someone needs to read it on each PR.

Author: this message has several goals 
- provide information for every new engineer
- remind everyone who create PRs to do some steps

so i don't think this repeatable PR to PR information is useless

> - [ ] After your pull request is merged, please add a message to the issue with instructions for how to test the fix or try the feature you added.

Reviewer: WOuld you mind adding an example of such message?

> - [ ] Please avoid closing the original issue before the release unless you are the author of the issue and want to close it or the fix/feature doesn't require the release procedure.

Reviewer: ```suggestion
- [ ] Do not close the original issue before the change is released. Please note, in some cases Github can automatically close the issue once PR is merged. Re-open the issue in such case.
```

### Describe Your Changes

Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.

### Checklist (Optional for External Contributions)

- [ ] Include a link to the GitHub issue in the commit message, if issue exists.
- [ ] Mention the change in the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md). Explain what has changed and why. If there is a related issue or documentation change - link them as well.

  Tips for writing a good changelog message::

    * Write a human-readable changelog message that describes the problem and solution.
    * Include a link to the issue or pull request in your changelog message.
    * Use specific language identifying the fix, such as an error message, metric name, or flag name.
    * Provide a link to the relevant documentation for any new features you add or modify.

> - [ ] After your pull request is merged, please add a message to the issue with instructions for how to test the fix or try the feature you added. Here is an [example](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4048#issuecomment-1546453726)

Reviewer: ```suggestion
- [ ] After your pull request is merged, please add a message to the original issue with instructions on how to test the change. See the message example [here](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4048#issuecomment-1546453726).
```

Reviewer: bump

Reviewer: bump
